tools/clang/unittests/HLSL/ValidationTest.cpp

```diff
@@ -4207,6 +4212,110 @@ TEST_F(ValidationTest, ValidateWithHash) {
   VERIFY_ARE_EQUAL(memcmp(Result, pHeader->Hash.Digest, sizeof(Result)), 0);
 }
 
+std::wstring GetEnvVarW(const std::wstring &VarName) {
+#ifdef _WIN32
+  if (const wchar_t *Result = _wgetenv(VarName.c_str()))
+    return std::wstring(Result);
+#else
+  std::string VarNameUtf8;
+  Unicode::WideToUTF8String(VarName.c_str(), &VarNameUtf8);
+  if (const char *Result = std::getenv(VarNameUtf8.c_str())) {
+    std::wstring ResultWide;
+    Unicode::UTF8ToWideString(Result, &ResultWide);
+    return std::wstring(ResultWide);
+  }
+#endif
+  return std::wstring();
+}
+
+void SetEnvVarW(const std::wstring &VarName, const std::wstring &VarValue) {
+#ifdef _WIN32
+  _wputenv_s(VarName.c_str(), VarValue.c_str());
+#else
+  std::string VarNameUtf8;
+  std::string VarValueUtf8;
+  Unicode::WideToUTF8String(VarName.c_str(), &VarNameUtf8);
+  Unicode::WideToUTF8String(VarValue.c_str(), &VarValueUtf8);
+  setenv(VarNameUtf8.c_str(), VarValueUtf8.c_str(), 1);
+#endif
```

Comment:
Are we planning to use this override behavior on linux?  I thought we only ran these validation tests on Windows.